model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/HistoryHandler.kt

```diff
@@ -82,45 +83,47 @@ class HistoryHandler(private val repositoriesManager: IRepositoriesManager) {
             get("/history") {
                 call.respondRedirect("../repos/")
             }
-            get("/history/{repoId}/{branch}") {
-                val repositoryId = RepositoryId(call.parameters["repoId"]!!)
-                val branch = repositoryId.getBranchReference(call.parameters["branch"]!!)
-                call.checkPermission(ModelServerPermissionSchema.branch(branch).pull)
-                val params = call.request.queryParameters
-                val limit = toInt(params["limit"], 500)
-                val skip = toInt(params["skip"], 0)
-                val latestVersion = repositoriesManager.getVersion(branch)
-                checkNotNull(latestVersion) { "Branch not found: $branch" }
-                call.respondHtmlTemplate(PageWithMenuBar("repos/", "../../..")) {
-                    headContent {
-                        style {
-                            unsafe {
-                                raw(
-                                    """
+            requiresLogin {
```

Comment:
The `requiresLogin` call verifies and loads the JWT token into the context of the call. Without it, the following permission checks will always fail because they don't who the user is.
Or in other words, `requiresLogin` does authentication and `checkPermission` does authorization.